java-checks/src/main/java/org/sonar/java/checks/DeadStoreCheck.java

```diff
@@ -176,12 +178,38 @@ private static void handleIdentifier(Set<Symbol> out, Set<Tree> assignmentLHS, I
 
   private void handleVariable(Set<Symbol> out, VariableTree localVar) {
     Symbol symbol = localVar.symbol();
-    if (localVar.initializer() != null && !out.contains(symbol)) {
-      createIssue(((VariableTreeImpl) localVar).equalToken(), localVar.initializer(), symbol);
+    ExpressionTree initializer = localVar.initializer();
+    if (initializer != null && !isUsualDefaultValue(initializer) && !out.contains(symbol)) {
```

Comment:
arguable but maybe passing the initializer and returning false if it is null could make things a bit clearer.